lib/addons/olm/index.ts

```diff
@@ -0,0 +1,71 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { Construct } from 'constructs';
+import { loadYaml } from "../../utils/yaml-utils";
+import { KubectlProvider, ManifestDeployment } from "../helm-addon/kubectl-provider";
+
+const OLMDownloadURL = "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/";
+
+export interface OlmAddOnProps {
+  /**
+  * OLM version
+  */
+  version?: string
+}
+
+/**
+ * Default OLM properties
+ */
+const defaultProps = {
+  version: "v0.21.2"
+};
+
+export class OlmAddOn implements ClusterAddOn {
+    readonly props: OlmAddOnProps;
+    readonly manifestUrls: string[];
+
+    constructor(props?: OlmAddOnProps) {
+      this.props = {...defaultProps, ...props};
+      this.manifestUrls = [
+        OLMDownloadURL+this.props.version+"/crds.yaml",
+        OLMDownloadURL+this.props.version+"/olm.yaml"
+      ];
+    }
+
+    deploy(clusterInfo: ClusterInfo): void | Promise<Construct> {
+      let previousResource: Construct;
+      const resources: Construct[] = [];
+      this.manifestUrls!.map((manifestUrl, urlIndex) => {
+        /* eslint-disable */
+        const request = require('sync-request');
```

Comment:
This loads the sync-request library in a cycle redefining the function. Can you move it out of the loop?

Reply:
yes this is bad, addressed.